pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -903,8 +904,8 @@ func applyUpstreamTLSSettings(env *model.Environment, cluster *apiv2.Cluster, tl
 			Sni:              tls.Sni,
 		}
 
-		// Fallback to file mount secret instead of SDS if meshConfig.sdsUdsPath isn't set or tls.mode is TLSSettings_MUTUAL.
-		if env.Mesh.SdsUdsPath == "" || tls.Mode == networking.TLSSettings_MUTUAL {
+		// Fallback to file mount secret instead of SDS if SDS is disabled globally, or meshConfig.sdsUdsPath isn't set or tls.mode is TLSSettings_MUTUAL.
```

Comment:
Why do we need this ? If a node has an SDS socket and sends it to pilot - what would be the reason for Pilot to not use it ? 

Can you write a small comment explaining when a user would do this ?

Reply:
today sidecar-injector uses ```.Values.global.sds.enabled``` to decide whether to inject sds uds socket mount to envoy container. Here is to make it consistent in pilot code whether to generate sdsconfig based on ``````.Values.global.sds.enabled``` ``` 

Later after we support per-pod rollout, both sidecar injector and pilot will add logic to check node.metadata.